src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,289 @@
+    PhotonCreator(
+        const SynchrotronFunctions::SyncFuncCursor& curF_1,
+        const SynchrotronFunctions::SyncFuncCursor& curF_2,
+        const uint32_t currentStep)
+            : curF_1(curF_1), curF_2(curF_2), randomGen(currentStep)
+    {
+        DataConnector &dc = Environment<>::get().DataConnector();
+        /* initialize pointers on host-side E-(B-)field databoxes */
+        FieldE* fieldE = &(dc.getData<FieldE > (FieldE::getName(), true));
+        FieldB* fieldB = &(dc.getData<FieldB > (FieldB::getName(), true));
+        /* initialize device-side E-(B-)field databoxes */
+        eBox = fieldE->getDeviceDataBox();
+        bBox = fieldB->getDeviceDataBox();
+    }
+
+    /** Initialization function on device
+     *
+     * \brief Cache EM-fields on device
+     *         and initialize possible prerequisites for ionization, like e.g. random number generator.
+     *
+     * This function will be called inline on the device which must happen BEFORE threads diverge
+     * during loop execution. The reason for this is the `__syncthreads()` call which is necessary after
+     * initializing the E-/B-field shared boxes in shared memory.
+     */
+    DINLINE void init(const DataSpace<simDim>& blockCell, const int& linearThreadIdx, const DataSpace<simDim>& localCellOffset)
+    {
+        /* caching of E and B fields */
+        cachedB = CachedBox::create < 0, ValueType_B > (BlockArea());
+        cachedE = CachedBox::create < 1, ValueType_E > (BlockArea());
+
+        /* instance of nvidia assignment operator */
+        nvidia::functors::Assign assign;
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldBBlock, bBox.shift(blockCell));
+        ThreadCollective<BlockArea> collective(linearThreadIdx);
+        collective(
+                  assign,
+                  cachedB,
+                  fieldBBlock
+                  );
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldEBlock, eBox.shift(blockCell));
+        collective(
+                  assign,
+                  cachedE,
+                  fieldEBlock
+                  );
+
+        /* wait for shared memory to be initialized */
+        __syncthreads();
+
+        /* initialize random number generator with the local cell index in the simulation*/
+        this->randomGen.init(localCellOffset);
+    }
+
+    DINLINE float_X emission_prob(
+        const float_X delta,
+        const float_X chi,
+        const float_X gamma,
+        const float_X mass,
+        const float_X charge) const
+    {
+
+        if(enableQEDTerm)
+        {
+            // quantum
+            const float_X z = float_X(2.0/3.0) * delta / ((float_X(1.0) - delta) * chi);
+
+            return DELTA_T * charge*charge * mass*SPEED_OF_LIGHT / (float_X(4.0)*PI*EPS0 * HBAR*HBAR) *
+                float_X(1.732050807) / (float_X(2.0) * PI) * chi/gamma * (float_X(1.0) - delta) / delta *
```

Comment:
offline discussion with @Heikman and @Flamefire:
if `delta_t` is too small `1/delta_t` will be `NaN`.

If the smalest number the random number generator produces, next to zero, is `EPSILON ~ 1e-7`, this should be save, since `1 / EPSILON ~ 1e7` can be represented in float. 

Please check the output of the random number generator.